File: src/main/java/org/cip4/xjdf/json/model/Schema.java
package org.cip4.xjdf.json.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

import java.util.List;
import java.util.Map;

@Data
@AllArgsConstructor
@Builder
public class Schema {
    private String $schema;
    private String $id;
    private String type;
    private Map<String, Schema> properties;
    private List<String> required;
    private Schema items;
    private String format;
    private List<String> enumValues;
    private Float minimum;
    private Float maximum;
    private String pattern;
    private List<Schema> oneOf;
    private Discriminator discriminator;
    private List<Schema> allOf;
    private List<Schema> anyOf;
    @NonNull
    @Builder.Default
    private Schemas schemas = new Schemas();
    private String $ref;
}

File: src/main/kotlin/org/cip4/xjdf/openapi/model/Schema.kt
package org.cip4.xjdf.openapi.model

import kotlinx.serialization.Serializable

@Serializable
data class Schema(
    var type: String? = null,
    var properties: Map<String, Schema>? = null,
    var required: List<String>? = null,
    var items: Schema? = null,
    var format: String? = null,
    var enum: List<String>? = null,
    var minimum: Int? = null,
    var maximum: Int? = null,
    var pattern: String? = null,
    var oneOf: List<Reference>? = null,
    var allOf: List<Schema>? = null,
    var minItems: Int? = null,
    var maxItems: Int? = null,
    val `$ref`: String? = null
) : YmlModel {

    companion object {
        fun ref(reference: String) = Schema(`$ref` = "#/components/schemas/$reference")
    }
}